### DISCUSSION TITLE
VPN Strongswan

### DISCUSSION THREAD

**Discussion by sokolblyat9** [strongswan/strongswan#1522] [General]
Hello!
I'm trying to set up a VPN using the instructions below:

https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-20-04-ru

I don’t understand how I can give a specific certificate with a key a specific ip address, that is, I want each client to have its own separate key / certificate, please tell me what I need to complete.
At the moment, I have reached the point that I can create 2 certificates and both will work (add another certificate in the leftcert line, separated by a comma, and so on, but I can’t set a static ip address)

**Comment by tobiasbrunner** [comment]
That tutorial apparently does not use certificates to authenticate clients, but username/password via EAP-MSCHAPv2. So you have to adapt the authentication settings (`rightauth`) and create certificates for your clients (using commands similar to step 3 but maybe with some changes, e.g. for the `--flag` and `--san` options, depending on your use case). Also see [this page](https://docs.strongswan.org/docs/5.9/pki/pkiQuickstart.html) in our docs.

> add another certificate in the leftcert line

Why do you need two local certificates?

> but I can’t set a static ip address

What do you mean?

**Reply by sokolblyat9** [reply]
I would like each client to have its own static IP address, that is, for example, the phone has 10.10.10.10/32, the TV has 10.10.10.11/32, the PC has 10.10.10.12/32, something like this

**Reply by tobiasbrunner** [reply]
That has nothing to do with certificates specifically (virtual IPs are assigned based on the client identities, so this works with any authentication method - certificates might even complicate things, depending on the client, see e.g. #1514). Have a look at the docs for [options to assign virtual IPs](https://docs.strongswan.org/docs/5.9/features/vip.html#_responder_configuration) (in ipsec.conf you could also add separate `conn` sections for each client identity/certificate and assign individual IPs via `rightsourceip`).

**Reply by sokolblyat9** [reply]
And what options do you have for setting up vpna?
for each option/setting type

**Reply by tobiasbrunner** [reply]
What do you mean?

**Reply by sokolblyat9** [reply]
How do I set up vpn with certificates? Do you have any detailed instructions?

**Reply by tobiasbrunner** [reply]
See e.g. the [quickstart examples](https://docs.strongswan.org/docs/5.9/config/quickstart.html).

**Reply by sokolblyat9** [reply]
one of these days I’ll try, unsubscribe, I’ll create everything exactly according to your official instructions, but you earned 1 minus for yourself, because you don’t know how to allocate a static internal IP address for each specific certificate, you stupidly didn’t even answer and didn’t ask for screenshots of what I could do wrong. Are you sure you created the software adequately ??? can you check it yourself??? to create some certificates and that to each certificate the konretny ah pi the address which is specified in any file was allocated???

**Reply by sokolblyat9** [reply]
> See e.g. the [quickstart examples](https://docs.strongswan.org/docs/5.9/config/quickstart.html).

which means moon, moon-net, sun,sun-net?
Why do we need astrology? Why it is impossible to adequately paint, roughly speaking, so to speak, a living example (for the test). Explain what these astrological names mean online, please.

**Reply by tobiasbrunner** [reply]
These are just some arbitrary names for the hosts in these examples and their networks. Did you not read the descriptions above the configs?

By the way, the names come from [our testing environment](https://docs.strongswan.org/docs/5.9/devs/testingEnvironment.html).

**Comment by satishkrishna687** [comment]
@sokolblyat9 since even I started with strongswan a month ago I would suggest you to see the configuration in the offical GitHub page itself. Just make sure the same CA certificate be available on both or all the machines and since you are using the IPsec.conf you can use the migration guide from swanctl.conf. Using the newer swanctl.conf is much easier. And thoroughly going through the swanctl.conf page in the documentation is highly recommended.

**Reply by sokolblyat9** [reply]
you can understand that in order to use the product, those who encounter it for the first time need adequate instructions, what it is, what it is for and how it is configured, you have information about this, but it is so scarce that in order to check the work service, you need to sort through different methods, there is no complete analysis of what will happen if I write numbers instead of an external IP address, internal ones. How could such a thing be done, it's terrible.
